stan/math/opencl/buffer_types.hpp

```diff
@@ -0,0 +1,83 @@
+#ifndef STAN_MATH_OPENCL_BUFFER_TYPES_HPP
+#define STAN_MATH_OPENCL_BUFFER_TYPES_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/constants.hpp>
+#include <CL/cl.hpp>
+
+namespace stan {
+namespace math {
+namespace opencl_kernels {
+
+// A read buffer will only add events to the read_write stack.
```

Comment:
```suggestion
// A read buffer will only add events to the read stack.
```

Reply:
Ty for the catch, this actually needs to be changed to

```
A read buffer will add events to the read and read_write stack
```